app/src/main/java/org/p2p/wallet/striga/wallet/repository/impl/StrigaWalletCachedRepository.kt

```diff
@@ -0,0 +1,76 @@
+package org.p2p.wallet.striga.wallet.repository.impl
+
+import timber.log.Timber
+import kotlin.reflect.KMutableProperty0
+import org.p2p.wallet.striga.common.model.StrigaDataLayerError
+import org.p2p.wallet.striga.common.model.StrigaDataLayerResult
+import org.p2p.wallet.striga.common.model.toSuccessResult
+import org.p2p.wallet.striga.user.repository.StrigaUserRepository
+import org.p2p.wallet.striga.wallet.models.StrigaCryptoAccountDetails
+import org.p2p.wallet.striga.wallet.models.StrigaFiatAccountDetails
+import org.p2p.wallet.striga.wallet.models.StrigaUserBankingDetails
+import org.p2p.wallet.striga.wallet.models.StrigaUserWallet
+import org.p2p.wallet.striga.wallet.models.ids.StrigaAccountId
+import org.p2p.wallet.striga.wallet.repository.StrigaWalletRepository
+
+internal class StrigaWalletCachedRepository(
+    private val remoteRepository: StrigaWalletRemoteRepository,
+    private val cache: StrigaWalletInMemoryRepository,
+    private val strigaUserRepository: StrigaUserRepository,
+) : StrigaWalletRepository {
+
+    private val timber: Timber.Tree = Timber.tag("StrigaWalletCachedRepository")
+
+    override suspend fun getUserWallet(): StrigaDataLayerResult<StrigaUserWallet> {
+        return withCache(cache::userWallet) {
+            remoteRepository.getUserWallet()
+        }
+    }
+
+    override suspend fun getFiatAccountDetails(
+        accountId: StrigaAccountId
+    ): StrigaDataLayerResult<StrigaFiatAccountDetails> {
+        return withCache(cache::fiatAccountDetails) {
+            remoteRepository.getFiatAccountDetails(accountId)
+        }
+    }
+
+    override suspend fun getCryptoAccountDetails(
+        accountId: StrigaAccountId
+    ): StrigaDataLayerResult<StrigaCryptoAccountDetails> {
+        return withCache(cache::cryptoAccountDetails) {
+            remoteRepository.getCryptoAccountDetails(accountId)
+        }
+    }
+
+    override suspend fun getAccountStatement(
+        accountId: StrigaAccountId,
+    ): StrigaDataLayerResult<StrigaUserBankingDetails> {
+        return withCache(null) {
```

Comment:
no need to cache now

Reply:
i mean why not `withCache() {` ?

Reply:
@kabramovp2p I think the argument should be set explicitly, because you might think that if you're not passing any argument, it will use some default variable

Reply:
yeah, absolutely 
you can see explicitly that this method uses no cache - so null is passed. 
```kotlin
withCache {
    ...
}
```
looks that there cache anyway

Reply:
@gslevinkov you know, I've just realized, why is there withCache called if the cache is not used at all? maybe this method should just return the result directly?
```
override suspend fun getAccountStatement(
     accountId: StrigaAccountId,
): StrigaDataLayerResult<StrigaUserBankingDetails> {
    val strigaUserFullName = strigaUserRepository.getUserDetails().unwrap()
        .userInfo
        .fullName
    return remoteRepository.getAccountStatement(accountId, strigaUserFullName)
}
```

Reply:
yeah, but we have unwrap here that will break the contract